config.go

```diff
@@ -131,10 +131,11 @@ func (c *config) discoverExternalComponents(path string) error {
 	if err != nil {
 		return err
 	}
-	for plugin := range pluginPaths {
+	for plugin, path := range pluginPaths {
```

Comment:
This change does work but it doesn't help make this code any more intuitive.  I believe that the original issue is that we're importing a module named "plugin", and then redeclaring "plugin" to be a variable inside of these loops. That's not something we ever should have done. I'd rather be very explicit and avoid namespace collisions. So instead of adding the path variable, and doing the same redeclaration inside the scope of the loop, let's just use variable names that aren't overloaded: 

example: 

```
for builderPlugin, builderPath := range pluginPaths {
```

I just tested, and if I use this instead I can drop the

```
plugin := plugin
path := path
```

Lines and the issue will still be fixed. Would you mind updating your PR with this change?  Otherwise I can just push a commit over your PR, if you'd rather.

Reply:
We could change the names of the variables for clarity, but I think you will still need to have that extra assignment to make it work for more than one plugin.

Reply:
I realized while I was renaming variables to make it easier to understand, that the "plugin" variable was not necessary because it was being used during the assignment, rather than later when the "func" would run.

Reply:
I still don't like using "plugin" at all here because of the namespace collision with the plugin module.

Reply:
And testing it with both a provisioner and a post-processor, I didn't need to assign a second time when I used unique variable names.

Reply:
> I still don't like using "plugin" at all here because of the namespace collision with the plugin module.

OK, I didn't realize that was a problem.  I will change that ;)

> And testing it with both a provisioner and a post-processor, I didn't need to assign a second time when I used unique variable names.

When I tried it with the example code <https://play.golang.org/p/8WrQ9AI5bPM> , the range variable (regardless of its name) overwrites the P1 and P2 variables, and thus the result is that with more than one plugin per type, you only get the last one.

Reply:
@SwampDragons I removed the reference to the "plugin" variable, saw what you meant by the namespace collision once I looked into it. I also found another reference in the discoverSingle() and removed that... does this work?

Reply:
Ah, I see. Yes, I like this way better. Thanks for implementing it!

Reply:
Do you want me to combine the commits?

Reply:
too late :)